### PR TITLE
GitHub pr status

### PR DESCRIPTION
**Pull request by sandro-h**

### CODE REVIEW

**Comment by sandro-h** [comment]
@dodie, @hosszubalazs: the interesting parts happen in `gong-notifier-github-pr-status`, the rest is just some small refactoring for the plugin settings.

The plugin is installed on gocd-int and I used it successfully in the she-github-pr pipeline to update the status of https://github.com/adnovum/gocd-gh-integration/pull/11.

There's also a README incl. user guide if you want more context: https://github.com/adnovum/gong-notifier/tree/github_pr_status/gong-notifier-github-pr-status

**Comment by sandro-h** [comment]
I think the deepcode integration was a trial by Marco. I dropped it now since we're not actively evaluating it and I don't feel like troubleshooting it in this context.

File: README.md [hunk 1]
```diff
@@ -8,7 +8,8 @@ Instead of each user specifying what they want to get notified about; or having
 managed by an admin, this plugin allows setting notification rules in the pipeline configuration via environment variables.
 
 The following plugins exist:
-- gong-notifier-email: Sends E-Mail notifications
+- [gong-notifier-email](gong-notifier-email): Sends E-Mail notifications
```
Comment (hosszubalazs) [inline README.md:11]: Did you check MarkDownlint?
https://github.com/DavidAnson/markdownlint

The reason I ask is I usually find that helpful, and the current READMEs actually look like compatible with those suggestions. :)

**Comment by hosszubalazs** [comment]
Retry DeepCode

**Comment by hosszubalazs** [comment]
> 
> 
> I think the deepcode integration was a trial by Marco. I dropped it now since we're not actively evaluating it and I don't feel like troubleshooting it in this context.

Since it looks like Gong is gaining traction, we could consider adding SonarCloud. Should not be a big issue to integrate into Gradle, and it would cool. Think of the badges you can put into the README. :)

**Comment by hosszubalazs** [comment]
Hey @sandro-h ,
supporting the merged builds, annotating the status of master, not just the PRs, is a feature we did not consider in the beginning but now realize the need for. GitHub actions supports this also. Very random repo as example:
https://github.com/actions/starter-workflows/branches

I think we do not really have a solution for this, as demonstrated the master is missing the green tick:
https://github.com/adnovum/gocd-gh-integration/branches

Do you think we can extend the functionality to support the merged builds too, possibly calling this `gong-notifier-github-**build**-status`  ?

Stuff we checked based on the int integration, and works well:
- pending and success status
- used a repo with anonymus access

Features still needing review:
- the failure status also happens as expected
- e-mailing still works as expected. 
- we only checked the gocd-gh-integration, which is a piublic repo. We plan to create a similar repo, but a private one, to help verify the authentication for the annotation.

**Comment by sandro-h** [comment]
> 
> 
> Hey @sandro-h ,
> supporting the merged builds, annotating the status of master, not just the PRs, is a feature we did not consider in the beginning but now realize the need for. GitHub actions supports this also. 

@hosszubalazs, yeah I think supporting that should be quite easy. I suggest to conclude this PR first and expand the functionality in a second PR.

File: README.md (hunk 1 above)
Comment (sandro-h) [reply README.md:12]: Thanks for the tip! I installed the VSCode extension and fixed all the issues in the READMEs.

**Comment by sandro-h** [comment]
> Since it looks like Gong is gaining traction, we could consider adding SonarCloud.
@hosszubalazs, I integrated SonarCloud and fixed the issues for this PR. I expect to fix more issues in the existing code once we merge this PR.

File: README.md (hunk 1 above)
Comment (hosszubalazs) [reply README.md:12]: Yeah I also use it form VSCode :)

**Comment by hosszubalazs** [comment]
Based on the current int GoCD install the plugin works for:
- private repos
- gong emailing still working
- failure status, not just success

Thanks for all the improvements like the READMEs and the Sonar integration meanwhile. I approve this change.

**Pull request merged**